### ISSUE TITLE
Multiple commands produce error with XCFramework 

### ISSUE THREAD

**Issue by TyrantDante** [CocoaPods/CocoaPods#10649]
* [x] I've read and understood the [*CONTRIBUTING* guidelines and have done my best effort to follow](https://github.com/CocoaPods/CocoaPods/blob/master/CONTRIBUTING.md).

# Report

## What did you do?
Hello we started to use xcframework recently in our project . I am having an issue with Cocoapods where we have library Reachability is generated with xcframework and trying to integrate with pod as dependency to other library after pod install while build we see and error like
```
Build system information
error: Multiple commands produce '/Users/xxxx/Library/Developer/Xcode/DerivedData/xxxx-ejsbkkdzaaghyjhcdxxsabbcjweq/Build/Products/Debug-iphoneos/XCFrameworkIntermediates/Reachability':
1) That command depends on command in Target 'Reachability-iOS11.0' (project 'Pods'): script phase “[CP] Copy XCFrameworks”
2) That command depends on command in Target 'Reachability-iOS13.0' (project 'Pods'): script phase “[CP] Copy XCFrameworks”
3) That command depends on command in Target 'Reachability-iOS14.0' (project 'Pods'): script phase “[CP] Copy XCFrameworks”
```

and podfile define like 
```
target 'xxx' do
    platform :ios, '11.0'
end
target 'xxx intents' do
    platform :ios, '13.0'
end
target 'xxx Desktop WidgetExtension' do
    platform :ios, '14.0'
end
```

## What did you expect to happen?
As you build it should copy respective xcframework like (Reachability ) xcframework

## What happened instead?
As you seen in above note getting same error

## CocoaPods Environment
```
   CocoaPods : 1.10.0
        Ruby : ruby 2.6.3p62 (2019-04-16 revision 67580) [universal.x86_64-darwin19]
    RubyGems : 3.0.3
        Host : Mac OS X 10.15.7 (19H2)
       Xcode : 12.4 (12D4e)
         Git : git version 2.26.1
Ruby lib dir : /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib
Repositories : 
               master - git - https://github.com/CocoaPods/Specs.git @ 344772c65b4c41c91851523f806b6d6922a3b69d
               trunk - CDN - https://cdn.cocoapods.org/
```

**Comment by dnkoutso** [comment]
This might be fixed by https://github.com/CocoaPods/CocoaPods/pull/10826

@TyrantDante can you try latest master branch or provide a sample app so I can verify?

**Comment by dnkoutso** [comment]
Confirmed fixed with 1.11.0.beta.
